### PR TITLE
Fix major issues after integration with the gateway

### PR DESCRIPTION
**Pull request by relu91**
I've started testing the wot-adapter inside the gateway. There were some issues about how we interpreted the lifecycle of a Device, now they should be mostly fixed. With this changes, I can confirm that thanks to this adapter it is possible to consume a W3C compliant WebThing and performing all the operations form WoT Interaction model

### CODE REVIEW

**Review by benfrancis — APPROVED** [review]

**Comment by benfrancis** [comment]
This needs @tim-hellhake's review in order to land.

File: src/wot-adapter.ts [hunk 1]
```diff
@@ -189,14 +204,21 @@ export class WoTAdapter extends Adapter {
...
       await wcd.save();
-
+      this.handleDeviceAdded(device);
       return device;
     }
   }
 
   handleDeviceSaved(_deviceId: string, _device: DeviceWithoutIdSchema): void {
     const device: WoTDevice = this.getDevice(_deviceId) as WoTDevice;
     device && device.start();
```
Comment (tim-hellhake) [inline src/wot-adapter.ts]: You can do this in typescript:
```suggestion
    device?.start();
```

Comment (tim-hellhake) [reply src/wot-adapter.ts]: Or better: move it into the if block :grimacing:

**Review by tim-hellhake — APPROVED** [review]

File: src/wot-adapter.ts (hunk 1 above)
Comment (relu91) [reply src/wot-adapter.ts]: Oh sure, sorry I missed it 👍🏻

**Comment by relu91** [comment]
Since we already have two approvals and I resolve a minor issue, I will go ahead and merge this.

**Pull request merged**